hypnogen/core/tts_result_cache.py

```
@@ -0,0 +1,81 @@
+logger = logging.getLogger(__name__)
+
+# Thread-safe cache storage
+_cache: Dict[str, tuple[np.ndarray, int]] = {}
+_cache_lock = threading.Lock()
+MAX_CACHE_SIZE = 100
+
+
+def _make_key(text: str, voice: str, speed: float) -> str:
+    """Create deterministic cache key."""
+    key_str = f"{text}|{voice}|{speed:.4f}"
+    return hashlib.md5(key_str.encode()).hexdigest()
```

Ruff (0.15.1)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)